common/lib/xmodule/xmodule/modulestore/split_mongo/split.py

```diff
@@ -1123,7 +1123,9 @@ def _block_matches_all(block_data):
             settings['children'] = qualifiers.pop('children')
         for block_id, value in course.structure['blocks'].iteritems():
             if _block_matches_all(value):
-                items.append(block_id)
+                # Check if the block is in the course tree
+                if self.has_path_to_root(block_id, course):
```

Comment:
Is there a performance implication for this change? Load testing would be a good idea.

Note that this could possibly also address orphaned discussion topics from showing up in the discussion tab (if we are calling this method...). FYI @adampalay